### DISCUSSION TITLE
How to get image from input type=file and send to server

### DISCUSSION THREAD

**Discussion by eatenpancreas** [leptos-rs/leptos#3134] [Q&A]
Hello, i've been trying to use Leptos to extract an image and send it in base64 (or any other proper encoding that works tbh) to a server function.

I followed this:

https://github.com/leptos-rs/leptos/discussions/2537

However i can't find a way to properly encode this in base64. I tried FileReader but you need an onLoad function and i have no idea to map that to a Leptos function call.

I know it's part a web_sys issue but i eventually need the callback to reach the leptos code again to send it to the server function. I hope anyone can help!

**Comment by gbj** [comment]
Since you mention a server function, just checking: have you looked at the [example of file uploads with server functions](https://github.com/leptos-rs/leptos/blob/main/examples/server_fns_axum/src/app.rs#L333)? IMO that is the best way to go, here.

Otherwise, you can do what you describe (getting the file, converting it to base64, and sending it as a string) but in that case Leptos does not provide any of the relevant APIs, and you're in `web-sys` or [`gloo-file`](https://docs.rs/gloo-file/latest/gloo_file/) territory.

**Comment by craigloewen-msft** [comment]
For any others who might find this, it is possible to do everything on the client side like so:

```

async fn PrintFileContent(input: Option<HtmlInputElement>) {
        log!("Input: {:?}", input);
        let value = input.unwrap().files();
        log!("files: {:?}", value);
        let value_unwrapped = value.unwrap();
        let get_file = value_unwrapped.get(0);
        log!("File option: {:?}", get_file);
        let file_text = get_file.unwrap().text();
        log!("File text: {:?}", file_text);
        let result = wasm_bindgen_futures::JsFuture::from(file_text).await;
        log!("Result: {:?}", result);
}

#[component]
fn SubtitleFileInput(translate_action: ServerAction<GetTranslateInfo>) -> impl IntoView {
    let file_input: NodeRef<Input> = NodeRef::new();

    view! {
        <h3>File Upload</h3>
        <input
            type="file"
            node_ref=file_input
            on:change=move |e| {
                let file_input_value = file_input.get();
                spawn_local(async move {
                    PrintFileContent(file_input_value).await;
                })
            }
        />
    }
}
```

You'll just need to make sure you add these crates as dependencies:
```
web-sys = { version = "0.3", features = ["FileList", "File"] } 
wasm-bindgen-futures = "0.4"
```
